nodejs/week2/mealSharing/src/bakckend/routes/meals.js

```diff
@@ -0,0 +1,83 @@
+const express = require("express");
+const router = express.Router();
+const meals = require("../data/reviewed-meals");
+
+router.get("/", (req, res) => {
+	const maxPrice = req.query.maxPrice;
+	const title = req.query.title;
+	const createdAfter = req.query.createdAfter;
+	const limit = req.query.limit;
```

Comment:
in general, all of these query parameters are optional and could be provided or not with the request. If they are provided then you can use them to filter the meals according to `maxPrice`, `title`, etc. But if they are not provided you just don't need to use them. e.g.

`/meals` => give me all meals
`/meals?maxPrice=100` => give me all cheap meals under 100 crowns
`/meals?title=chicken` => give me all meals with chicken
`/meals?title=chicken&maxPrice=100` => give me all cheap meals with chicken

When the service behaves in this way in can be really useful because it provides a very rich functionality for the user.

Reply:
I don't think if it was clear above but, in this function you're only handling one single query param:

e.g. if you provide `maxPrice` that's the only one that applies

You need to be able to combine all the query params and have them filter the meals.